app/org/sagebionetworks/bridge/dynamodb/DynamoUserConsentDao.java

```diff
@@ -89,6 +89,7 @@ public UserConsent getActiveUserConsent(String healthCode, SubpopulationGuid sub
 
         DynamoDBQueryExpression<DynamoUserConsent3> query = new DynamoDBQueryExpression<DynamoUserConsent3>()
             .withScanIndexForward(false)
+            .withLimit(1)
             .withQueryFilterEntry("withdrewOn", new Condition().withComparisonOperator(ComparisonOperator.NULL))
             .withHashKeyValues(hashKey);
```

Comment:
Okay, I created a test that directly creates two consent records for one subpopulation, and verified that withdrawing didn't work correctly. Then I fixed it by iterating through all consents and withdrawing all of them.